simulator/cmd/simulator/simulator.go

```diff
@@ -94,7 +95,7 @@ func startSimulator() error {
 		return xerrors.Errorf("kubeapi-server is not ready: %w", err)
 	}
 
-	dic, err := di.NewDIContainer(client, dynamicClient, restMapper, etcdclient, restCfg, cfg.InitialSchedulerCfg, cfg.ExternalImportEnabled, cfg.ResourceSyncEnabled, importClusterResourceClient, importClusterDynamicClient, cfg.Port, syncer.Options{})
+	dic, err := di.NewDIContainer(client, dynamicClient, restMapper, etcdclient, restCfg, cfg.InitialSchedulerCfg, cfg.ExternalImportEnabled, cfg.ResourceSyncEnabled, importClusterResourceClient, importClusterDynamicClient, cfg.Port, syncer.Options{}, resourceapplier.Options{})
```

Comment:
Why do we need `resourceapplier.Options{}`? 
If you meant to use it for oneshot importer, then that's fine.
But, if you create it just in case someone wants, then we don't need it until "someone" comes to us and makes a feature request.

Reply:
In the current implementation, `syncer` filters and mutates resources to apply and the filters and mutations can be configurable with the option. This PR moves these logic to `resourceapplier` so we need `resourceapplier`'s option to configure them.

Another possible way is callers of `resourceapplier` (`syncer`, `oneshotimporter`, `replayer` or etc) filter and mutate resources and it just applies them. But all the callers would need filters and mutations so I think it's better to commonize them as `resourceapplier`.

Reply:
Ah, okay I forgot about the existing options. Then you need to tweak this doc.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/simulator/docs/import-cluster-resources.md?plain=1